v2/relay/ipcs.go

```diff
@@ -0,0 +1,110 @@
+package relay
+
+import (
+	"errors"
+	"net"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+
+	asnutil "github.com/libp2p/go-libp2p-asn-util"
+	ma "github.com/multiformats/go-multiaddr"
+	manet "github.com/multiformats/go-multiaddr/net"
+)
+
+var (
+	ErrNoIP              = errors.New("no IP address associated with peer")
+	ErrTooManyPeersInIP  = errors.New("too many peers in IP address")
+	ErrTooManyPeersInASN = errors.New("too many peers in ASN")
```

Comment:
Yeah, the only reason I exported them is so that the test could could live together with the other tests in the `test` directory.
It felt odd to have just one test in here.